feat: add HLS 2.0 dataset

hrodmn opened the pull request:
HLS imagery in VEDA!
![image](https://github.com/user-attachments/assets/955f3e95-ab3a-46b6-a88f-f83fa0d7e946)

The new layers are dependent on changes from https://github.com/NASA-IMPACT/veda-ui/pull/1534 so I updated the git ref for ./.veda/ui to enable the layers in the deploy preview. If we need to undo that change before merging let me know.

I pointed the .mdx file at the STAC collection records in the [CMR LPCLOUD STAC](https://cmr.earthdata.nasa.gov/stac/LPCLOUD/) - if we want to add collection records to the VEDA STAC instead, I can do that!

### Adding a Dataset Overview:
- [ ] Dataset is available in the production VEDA STAC 
- [x] Production STAC url is referenced in the dataset overview
- [x] All metadata, including those for the E&A page, is completed in the mdx file
- [x] Any images are compressed, ideally <500 KB
- [ ] At least one code owner has reviewed the PR

hanbyul-here commented on package.json:
```diff
@@ -39,6 +39,7 @@
     "@trussworks/react-uswds": "^9.1.0",
     "@uswds/uswds": "^3.8.1",
     "autoprefixer": "^10.4.19",
+    "date-fns": "^4.1.0",
```
is this necessary?

hrodmn replied:
Possibly not. I was hitting errors about that library being missing so I added it, but that might have been related to my `yarn serve` troubles. I'll try removing it and re-testing.

hanbyul-here replied:
Yah, all the dependencies should be supplied by veda ui so this should not be necessary!

hanbyul-here commented on docs/SETUP.md:
````diff
@@ -56,10 +56,10 @@ Get your Mapbox access token from Mapbox Dashboard. Put the key in `.env.local`
 
 ```
-yarn serve
+./.veda/veda serve
````
I thought they are the same thing! Am I misunderstanding?

hrodmn replied:
I was having some trouble with the hot reload capability when I was running `yarn serve` - after saving a file it would not properly reload the browser preview. Then I found the `./.veda/veda` script which appears to add some modifications to all of the basic `yarn` commands including `serve`. After I started using `./.veda/veda serve` the dev experience was much better!

hanbyul-here commented:
Am I supposed to be able to browse the data through preview? Nothing seems to load for me

hrodmn commented:
> Am I supposed to be able to browse the data through preview? Nothing seems to load for me

The experience is going to be a bit different than for most other collections because `minzoom` is set to 7, which requires you to zoom in quite a bit. It would be cool to add a UI element that tells us when we are outside of a layer's `minzoom`. 

Another thing that is different from other collections is that the data are available in satellite orbit fashion, which means data for an AOI will not be available every day but only for days where the satellite flew over. For HLS data you might need to try a few different days (probably 2-3 days) before you find anything. It might also have more of a processing lag so you could need to go a few days back from today's date before finding the first available data. Unfortunately that detail is not captured in the STAC collection's temporal extent.

hanbyul-here commented:
Ah! ok, I could populate the dataset, very very cool! : https://deploy-preview-801--visex.netlify.app/exploration?datasets=%5B%7B%22id%22%3A%22HLSL30_2.0_false_color%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Average%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22St+Deviation%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%2C%7B%22id%22%3A%22HLSL30_2.0_true_color%22%2C%22settings%22%3A%7B%22isVisible%22%3Afalse%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Average%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22St+Deviation%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%2C%7B%22id%22%3A%22HLSS30_2.0_false_color%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Average%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22St+Deviation%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%5D&taxonomy=%7B%7D&search=&date=2024-08-30T15%3A00%3A00.000Z&aois=%5B%5D&dateRange=

I wonder if people can figure out how to navigate the dataset. I would have just given up if I hadn't had a clue from your screenshot. Of course, nothing on this PR is the problem. It is more like the dashboard is not prepared very well for this type of dataset (sparse spatially and temporarily). I wish we could have a better indicator to guide users,  but I can't think of a quick way to do it now. (Actually, I think the dataset should be zoomed-in automatically at least but it is not doing that now. I will look at it.)

But I am not a very avid user of hls dataset - do you think hls users expect this type (navigating to specific time and orbits) of experience? 

I am nto sure any of this should be a blocker for this PR - I will leave this up to @anayeaye and @slesaad  .

anayeaye commented:
@hanbyul-here @hrodmn  user education about the temporal complexity of satellite data is what drove the decision to just use specific granules for the HLS events stories--it isn't intuitive without an illustration of how the data are captured and why they are not globally available for a date. I wonder if there is a way to default to searching a 9 or 16 day range of dates for these collections? The titiler would handle mosaic at that level (although probably slowly on a global view). I think there would be another place for confusion in the analysis view. I agree it shouldn't block the PR but I also think we will see bug reports.

1. Can the date selection be defaulted to a window instead of a single date for a collection?
2. Can explore be fixed to a tighter zoom for a given collection?

hrodmn commented:
> @hanbyul-here @hrodmn user education about the temporal complexity of satellite data is what drove the decision to just use specific granules for the HLS events stories--it isn't intuitive without an illustration of how the data are captured and why they are not globally available for a date. 

I completely agree that the interface of daily orbits is not very intuitive and can be discouraging if you start browsing a map and nothing appears at first. Is there precedent for a pop-up with some information for the user when a specific layer is selected?

> 1. Can the date selection be defaulted to a window instead of a single date for a collection?

I am not positive about the veda-config/ui side of this but I did an [analysis of tile rendering performance](https://developmentseed.org/titiler-cmr/dev/datasets/hls_tiling/) various time windows and zoom levels. TL;DR: extending the `datetime` range to multiple days is possible but the tiles will render much more slowly. It could improve the experience by showing more imagery, but could make for a worse experience because of rendering lag.

> 2. Can explore be fixed to a tighter zoom for a given collection?

This could be cool! If a user navigates to the map from the HLS catalog entry maybe we could snap to a date/zoom where we know there is data. 

Another idea: some kind of hint in the UI when the layer's minzoom is higher (e.g. 7 in this case) than the map's current zoom level would go a long way here. Like "zoom in to see data from the HLS layer".

hanbyul-here commented:
Thanks for more info and input 🙇  @hrodmn  @anayeaye 

We usually get the metadata of the dataset (ex. where the datasets are, what the minimum zoom is...) from stac 🤔 But non-rastertimeseries data, such as HLS, doesn't go through stac requests. Where should we get the metadata of the datasets for HLS? 

Also, I do not want to drag the PR, especially when we do not have a specific action plan to improve the experience. Where are we on this PR? Do you think we are comfortable merging this PR with the caveats mentioned?

hrodmn commented:
Thanks for the nudge @hanbyul-here 

> We usually get the metadata of the dataset (ex. where the datasets are, what the minimum zoom is...) from stac 🤔 But non-rastertimeseries data, such as HLS, doesn't go through stac requests. Where should we get the metadata of the datasets for HLS?

If we make a placeholder collection record in the VEDA STAC with a `render` config that declares `minzoom=7` will VEDA UI use it to tell users to zoom in if they are zoomed too far out (e.g. `zoom=3`)? Right now the `minzoom` setting from the layer config is getting passed to titiler-cmr but there is no UI element that tells the user about the minimum zoom.

> Do you think we are comfortable merging this PR with the caveats mentioned?

While the user browsing experience may not be optimal right now, the new HLS layers could be useful in the context of new data stories where specific zoom + locations are pre-set to match the available imagery. If the improvements we want to make for usability will require veda-ui changes I think we should merge this and open a ticket for the UI changes. If there is something else we could add to this veda-config PR to improve the experience I am happy to add to it!

hanbyul-here commented:
Ok! let's move this PR forward with - removing unnecessary changes from this PR (dependencies, UI change) - filing issues mentioned in this thread @hrodmn !

hrodmn commented:
@hanbyul-here I removed the changes to the dependencies and veda-ui submodule and filed a follow-up ticket for HLS improvements: https://github.com/NASA-IMPACT/veda-config/issues/835.

hanbyul-here reviewed (changes requested):
Can you undo the submodule change before merge?

hrodmn commented:
> Can you undo the submodule change before merge?

Ah, sorry about that. I think I updated it to match `main` not `develop`. I rebased this branch on `develop` so I think it's good now.

hanbyul-here reviewed (approved):
Thank you for working on this !!
